File: newton.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double newtonRaphson(double t);

int main()
{
  double t;
  puts("Please enter a double number:\n");
  scanf("%lf", &t);

  double r = newtonRaphson(t);
  printf("\nThe root of %5.2f = %f.\n", t,r);
}

double newtonRaphson(double t)
{
  double epsilon = 0.0000001;
  double root    = (t+0)/2.0;
  while( fabs(root*root-t)>=epsilon ){
    root = (root + t/root)/2.0;
  }
  return root;
}

File: newton.java
public class newton
{
    public static void main(String[] args)
    {
        double t = Double.parseDouble(args[0]);
	double r = newtonRaphson(t);
        System.out.println("The root of " + t + " is " + r);
    }
    
    public static double newtonRaphson(double t)
    {
        if (t<0) return -1;
	
	double epsilon = 0.000001;
	double root;
	root = (0 + t)/2;
	while ( Math.abs(root*root-t) >= epsilon){
            root = (root+t/root)/2.0;
	}

	return root;
    }
}   
